cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto

```diff
@@ -12,124 +12,249 @@ import "protoc-gen-openapiv2/options/annotations.proto";
+message GetAvailablePackageDetailResponse {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{"available_package_detail": [{}]}'
+  // };
   
-  // PackageRepositoryReference
-  //
-  // A PackageRepository is identified by a fully qualified name and
-  // optionally a specific namespace.
-  message PackageRepositoryReference {
-    // Package repository name
-    //
-    // The fully qualified name of the package repository.
-    string name = 1;
-
-    // Package repository namespace
-    //
-    // An optional namespace for namespaced package repositories.
-    string namespace = 2;
-  }
-  // Package repository reference
+  // Available package detail
   //
-  // An optional package repository reference where this package is located. Not
-  // all plugins may support this back reference (eg. kapp-controller)
-  PackageRepositoryReference repository = 4;
-
-  // Package repository plugin
-  //
-  // The plugin used to interact with this package repository.
-  kubeappsapis.core.plugins.v1alpha1.Plugin plugin = 5;
+  // List of AvailablePackageDetail
+  AvailablePackageDetail available_package_detail = 1;
 }
 
-// GetAvailablePackages
+// GetPackageRepositories
 //
-// Response for GetAvailablePackages
-message GetAvailablePackagesResponse {
+// Response for GetPackageRepositories
+message GetPackageRepositoriesResponse {
   option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
-    example: '{"packages": [{"name": "app1.example.com", "version": "9.10.09", "iconUrl": ""}]}'
+    example: '{"repositories": [{"name": "repo-name.example.com", "namespace": "", "url": "foo.registry.example.com/repo-name/main@sha256:cecd9b51b1f29a773a5228fe04faec121c9fbd2969de55b0c3804269a1d57aa5"}]}'
   };
-  // List of AvailablePackage
-  repeated AvailablePackage packages = 1;
+
+  // Repositories
+  //
+  // List of PackageRepository
+  repeated PackageRepository repositories = 1;
 }
 
-// GetPackageRepositories
+// -- Start definitions used in the responses  -- 
+
+// AvailablePackageSummary
 //
-// Request for GetPackageRepositories
-message GetPackageRepositoriesRequest {
-  // A cluster name can be provided if multiple clusters are configured,
-  // otherwise the current cluster will be assumed.
-  string cluster = 1;
+// An AvailablePackageSummary provides a summary of a package available for installation
+// useful when aggregating many available packages.
+message AvailablePackageSummary {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
 
-  // A namespace can be provided if the package repositories for a specific namespace
-  // are requested, when supported by the packaging format.
-  string namespace = 2;
+  // Available package reference
+  //
+  // A reference uniquely identifying the package.
+  AvailablePackageReference available_package_ref = 1;
+  
```

Comment:
We'll need to add a:
```
string name = 1;
```
here since the available package ref no longer (necessarily) stores the name, but rather an identifier.

Reply:
Great, thanks, I'm changing, for consistency with other msg, it to

```proto
  AvailablePackageReference available_package_ref = 1;
  string name = 2;
```